File: integration_tests/workspaces/python/callchain.py
"""Call chain module for testing call hierarchy."""


def leaf_function() -> str:
    """A leaf function with no outgoing calls."""
    return "leaf result"


def middle_function() -> str:
    """A function that calls leaf_function."""
    result = leaf_function()
    return f"middle: {result}"


def entry_point() -> None:
    """An entry point that calls middle_function."""
    msg = middle_function()
    print(msg)


def another_caller() -> None:
    """Another caller of middle_function."""
    print(middle_function())

File: integration_tests/workspaces/rust/src/callchain.rs
// Call chain module for testing call hierarchy

pub fn leaf_function() -> String {
    String::from("leaf result")
}

pub fn middle_function() -> String {
    let result = leaf_function();
    format!("middle: {}", result)
}

pub fn entry_point() {
    let msg = middle_function();
    println!("{}", msg);
}

pub fn another_caller() {
    println!("{}", middle_function());
}
